### PR TITLE
Updated to be compatible with finagle 6.27.0

### PR DESCRIPTION
**Pull request by mkhq**

### CODE REVIEW

File: src/main/scala/com/twitter/finagle/websocket/WebSocketHandler.scala [hunk 1]
```diff
@@ -10,11 +9,38 @@ import org.jboss.netty.handler.codec.http.websocketx._
...
 import scala.collection.JavaConversions._
 
+private[finagle] class CancelledException extends Exception
+
 class WebSocketHandler extends SimpleChannelHandler {
   protected[this] val messagesBroker = new Broker[String]
   protected[this] val binaryMessagesBroker = new Broker[Array[Byte]]
   protected[this] val closer = new Promise[Unit]
 
+  implicit class RichChannelFuture(future: ChannelFuture) {
```
Comment (travisbrown) [inline src/main/scala/com/twitter/finagle/websocket/WebSocketHandler.scala]: Since this is only used in a couple of places, it might be a little cleaner to define conversion methods that are used directly.

File: src/main/scala/com/twitter/finagle/websocket/WebSocketHandler.scala [hunk 2]
```diff
@@ -10,11 +9,38 @@ import org.jboss.netty.handler.codec.http.websocketx._
...
+          promise.setValue(())
+        else if(cf.isCancelled)
+          promise.setException(new CancelledException)
+        else
+          promise.setException(cf.getCause)
+      }
+    })
+
+
+    def toFuture:Future[Unit] = {
```
Comment (travisbrown) [inline src/main/scala/com/twitter/finagle/websocket/WebSocketHandler.scala]: This is super nitpicky, but can we have a space after the colon and before the type for the sake of consistency?

**Comment by travisbrown** [comment]
Thanks, @mkhq—looks good to me except for a couple of little things. Any thoughts, @sprsquish?

File: src/main/scala/com/twitter/finagle/websocket/WebSocketHandler.scala (hunk 1 above)
Comment (sprsquish) [inline src/main/scala/com/twitter/finagle/websocket/WebSocketHandler.scala]: I've fallen behind the state of Finagle's art. Why is this necessary? Is Finagle not providing something. Plus the exception will now be a finagle-websocket exception (as opposed to a Finagle cancelled exception). Is that necessary?

**Comment by mkhq** [comment]
@travisbrown sorry for the delay, I made the implicit class into a function and used a CancelledRequestException from finagle instead of a custom exception
